src/EPR.Calculator.Frontend/Controllers/ParameterUploadFileController.cs

```diff
@@ -90,7 +90,7 @@ private async Task<string> GetViewName(IFormFile fileUpload)
             var schemeTemplateParameterValues = await CsvFileHelper.PrepareSchemeParameterDataForUpload(fileUpload);
 
             this.ViewData["schemeTemplateParameterValues"] = schemeTemplateParameterValues.ToArray();
-            this.TempData["FileName"] = fileUpload.FileName;
+            this.HttpContext.Session.SetString("FileName", fileUpload.FileName);
```

Comment:
Please create constants for session variables
Also, you need to clean up all the values in try/catch block. Thanks